administrator/components/com_jedchecker/libraries/rules/xmlinfo.php

```diff
@@ -70,51 +92,210 @@ public function check()
+		if ($type === 'plugin' && isset($xml['group']) && strpos($extension, 'plg_') !== 0)
+		{
+			$extension = 'plg_' . $xml['group'] . '_' . $extension;
 		}
 
 		// Load the language of the extension (if any)
 		$lang = JFactory::getLanguage();
-		$lang->load($folder_name,$folder_path);
 
-		// Failed to parse the xml file.
-		// Assume that this is not a extension manifest
-		if (!$xml)
+		// search for .sys.ini translation file
+		$lang_dir = dirname($file);
+		$lang_tag = 'en-GB'; // $lang->getDefault();
+
+		// Populate list of directories to look for
+		$lookup_lang_dirs = array();
+
+		if (isset($xml->administration->files['folder']))
 		{
-			return true;
+			$lookup_lang_dirs[] = trim($xml->administration->files['folder'], '/') . '/language/' . $lang_tag;
 		}
 
-		// Check if this is an extension manifest
-		// 1.5 uses 'install', 1.6 uses 'extension'
-		if ($xml->getName() != 'install' && $xml->getName() != 'extension')
+		if (isset($xml->files['folder']))
+		{
+			$lookup_lang_dirs[] = trim($xml->files['folder'], '/') . '/language/' . $lang_tag;
+		}
+
+		$lookup_lang_dirs[] = 'language/' . $lang_tag;
+
+		if (isset($xml->administration->languages))
+		{
+			$folder = trim($xml->administration->languages['folder'], '/');
+
+			foreach ($xml->administration->languages->language as $language)
+			{
+				if (trim($language['tag']) === $lang_tag)
+				{
+					$lookup_lang_dirs[] = trim($folder . '/' . dirname($language), '/');
+				}
+			}
+		}
+
+		if (isset($xml->languages))
 		{
-			return true;
+			$folder = trim($xml->languages['folder'], '/');
+
+			foreach ($xml->languages->language as $language)
+			{
+				if (trim($language['tag']) === $lang_tag)
+				{
+					$lookup_lang_dirs[] = trim($folder . '/' . dirname($language), '/');
+				}
+			}
+		}
+
+		$lookup_lang_dirs[] = '';
+
+		$lookup_lang_dirs = array_unique($lookup_lang_dirs);
+
+		// Looking for language file in specified directories
+		foreach ($lookup_lang_dirs as $dir)
+		{
+			$lang_sys_file =
+				$lang_dir . '/' .
+				($dir === '' ? '' : $dir . '/') .
+				$lang_tag. '.' . $extension . '.sys.ini';
+			if (is_file($lang_sys_file))
+			{
+				$loadLanguage = new ReflectionMethod($lang, 'loadLanguage');
+				$loadLanguage->setAccessible(true);
+				$loadLanguage->invoke($lang, $lang_sys_file, $extension);
+				break;
+			}
 		}
 
 		// Get the real extension's name now that the language has been loaded
-		(string) $extension_name = $lang->_($xml->name);
+		$extension_name = $lang->_((string) $xml->name);
 
 		$info[] = JText::sprintf('COM_JEDCHECKER_INFO_XML_NAME_XML', $extension_name);
 		$info[] = JText::sprintf('COM_JEDCHECKER_INFO_XML_VERSION_XML', (string) $xml->version);
 		$info[] = JText::sprintf('COM_JEDCHECKER_INFO_XML_CREATIONDATE_XML', (string) $xml->creationDate);
+
 		$this->report->addInfo($file, implode('<br />', $info));
 
+		// NM3 - Listing name contains “module” or “plugin”
+		if (preg_match('/\b(?:module|plugin)\b/i', $extension_name))
```

Comment:
This regex only targets seems to narrow, it should instead look something like this which is more greedy.
```php
if (preg_match('/(module|plugin)/i', $extension_name))
```
Here is the idea https://regexr.com/5p86l

Let me know if this is to greedy ;)

Reply:
Initially I wasn't sure about other English words containing "module" or "plugin" as a substring, and `\b`s were added to avoid possible false-positives. But I've just checked at https://github.com/dwyl/english-words that it seems to be safe to check it "greedy". I'll commit this change a little bit later.

Reply:
Done

Reply:
That is a valid point... I will get back to you on this, maybe its to greedy now. I need to check the terms and make sure we don't make it to strict.

Reply:
This is the list of banned words for extensions that are checked when the extension is submitted: 

- module
- plugin
- component
- free
- extension
- template

Any match of these words is rejected.

Also the word "Joomla" can't be in an extension name unless it is included at the end of the name " for Joomla"

Reply:
One more, the extension name can't start with "mod_".

Reply:
I have an idea: let's trigger error for exact match (e.g. "Super Module"), warning for leading/trailing part of the word (e.g. "Supermodule"), and none (or a notice) for inner part of the word (e.g. "demoduled"). Maybe with exception of "free", because there are many words containing or ending with "free", and it's better to check for just an exact matching. What do you think?

@anibalsanchez Does JED allow the "package" word in the title? What about "plugin" written as "plug-in"? And what about other prefixes (com_, plg_, tpl_, pkg_)?

BTW, does JED allow emoji in extension name? (or "no longer" 😊?)